### PR TITLE
RandomIntegers fixes

### LINKED ISSUE #1379: [DEFECT] randomIntegers
--------
Defect Description
--------

**Describe the defect**

##### What did you expect to see happen?
In randomUtils, the randomIntegers method should generate random integers that are uniformly distributed

##### What did you see instead?
The generated random integers are not uniformly distributed

##### Do you have a suggested fix for the development team?
Edit lines 191-192 as follows:
    intRange = high - low **+ 1.0**
    rawNum = low + random(engine=engine)*intRange **- 0.5**

**Describe how to Reproduce**
1. 
2. 
3. 
4. 

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

**Platform (please complete the following information):**
 - Version: [e.g. 22]
 - Dependencies Installation: [CONDA or PIP]

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by mandd**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1379 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by mandd** [comment]
Work performed on the tests:
[tests.xlsx](https://github.com/idaholab/raven/files/5598695/tests.xlsx)

**Comment by mandd** [comment]
Test with 10K samples for 'randomUtils.randomIntegers(1,4,None,engine=None)'
test10K

File: framework/utils/randomUtils.py [hunk 1]
```diff
@@ -188,8 +188,8 @@ def randomIntegers(low, high, caller=None, engine=None):
   if isinstance(engine, np.random.RandomState):
     return engine.randint(low, high=high+1)
   elif isinstance(engine, findCrowModule('randomENG').RandomClass):
-    intRange = high - low
-    rawNum = low + random(engine=engine)*intRange
+    intRange = high - low + 1.0
+    rawNum = low + random(engine=engine)*intRange - 0.5
     rawInt = int(round(rawNum))
```
Comment (joshua-cogliati-inl) [inline framework/utils/randomUtils.py:193]: Maybe we should switch this to floor or ceiling instead of round and that might simplify the previous lines.

Comment (joshua-cogliati-inl) [reply framework/utils/randomUtils.py:193]: as in ```rawInt = math.floor(rawNum)``` and then I think the above can be ```rawNum = low + random(engine=engine)*intRange```

Comment (joshua-cogliati-inl) [reply framework/utils/randomUtils.py:193]:
```
>>> import math
>>> low = 3
>>> high = 6
>>> intRange = high - low + 1.0
>>> intRange
4.0
>>> low + 0*intRange
3.0
>>> math.floor(low + 0*intRange)
3
>>> math.floor(low + 0.99999*intRange)
6
>>> low + 0.99999*intRange
6.99996
>>> 
```

Comment (mandd) [reply framework/utils/randomUtils.py:193]: Your method is a complementary approach. I am running your edits agianst the tests,

**Comment by mandd** [comment]
Comments have been addressed

File: framework/utils/randomUtils.py [hunk 2]
```diff
@@ -188,9 +189,9 @@ def randomIntegers(low, high, caller=None, engine=None):
   if isinstance(engine, np.random.RandomState):
     return engine.randint(low, high=high+1)
   elif isinstance(engine, findCrowModule('randomENG').RandomClass):
-    intRange = high - low
+    intRange = high - low + 1.0
     rawNum = low + random(engine=engine)*intRange
-    rawInt = int(round(rawNum))
+    rawInt = math.floor(rawNum)
```
Comment (joshua-cogliati-inl) [inline framework/utils/randomUtils.py:194]: Looks good to me.

**Review by alfoa — APPROVED** [review]
Modifications approved. As Chair of the CCB, I approve the modifications of the requirements tests. @mandd please, send an email to the user list to inform about the issue and its addressing.

**Comment by alfoa** [comment]
Checklist passed...

Merging...

**Pull request merged**